webapp/components/survey/CategoryDetails/store/actions/category/useInit.js

```diff
@@ -1,54 +1,28 @@
 import { useCallback } from 'react'
-import { useDispatch } from 'react-redux'
-import axios from 'axios'
 
-import * as A from '@core/arena'
-import * as Category from '@core/survey/category'
+import { matchPath, useLocation } from 'react-router'
 
-import { SurveyActions, useSurveyId } from '@webapp/store/survey'
+import { designerModules, appModuleUri } from '@webapp/app/appModules'
 
-import { useFetchLevelItems } from '../item/useFetchLevelItems'
-import { State } from '../../state'
-
-const _createCategory = async ({ surveyId, onCategoryCreated, dispatch }) => {
-  const {
-    data: { category },
-  } = await axios.post(`/api/survey/${surveyId}/categories`, Category.newCategory())
-
-  if (onCategoryCreated) {
-    onCategoryCreated(category)
-  }
-  dispatch(SurveyActions.metaUpdated())
-
-  return category
-}
+import * as API from '@webapp/service/api'
 
-const _fetchCategory = async ({ surveyId, categoryUuid }) => {
-  const {
-    data: { category },
-  } = await axios.get(`/api/survey/${surveyId}/categories/${categoryUuid}?draft=true&validate=true`)
+import { useSurveyId } from '@webapp/store/survey'
 
-  return category
-}
+import { useFetchLevelItems } from '../item/useFetchLevelItems'
+import { State } from '../../state'
 
 export const useInit = ({ setState }) => {
-  const dispatch = useDispatch()
   const surveyId = useSurveyId()
   const fetchLevelItems = useFetchLevelItems({ setState })
+  const { pathname } = useLocation()
 
-  return useCallback(async ({ state, onCategoryCreated }) => {
-    const categoryUuid = State.getCategoryUuid(state)
-    let category = null
+  return useCallback(async ({ categoryUuid }) => {
+    const category = await API.fetchCategory({ surveyId, categoryUuid })
 
-    if (A.isEmpty(categoryUuid)) {
-      category = await _createCategory({ surveyId, onCategoryCreated, dispatch })
-    } else {
-      category = await _fetchCategory({ surveyId, categoryUuid })
-    }
+    const inCategoriesPath = Boolean(matchPath(pathname, `${appModuleUri(designerModules.category)}:uuid/`))
 
-    const stateUpdated = State.assocCategory({ category })(state)
+    const stateUpdated = State.create({ category, categoryUuid, inCategoriesPath })
```

Comment:
I don't think we need the categoryUuid in the state anymore. Once fetched in init, the categoryUuid will be the category.uuid